samples/GalleryApp/MainPage.xaml.cs

```diff
@@ -21,29 +16,80 @@ namespace GalleryApp
     /// </summary>
     public sealed partial class MainPage : Page
     {
+        public ObservableCollection<Skill> SkillPages { get; } = new ObservableCollection<Skill>();
+
+        private static List<SkillCategory> allCategories;
+        private static SemaphoreSlim _semaphore = new SemaphoreSlim(1);
+
         public MainPage()
         {
             this.InitializeComponent();
         }
 
         /// <summary>
-        /// Navigation to Skeletal Detector Page
+        /// Invoked when the Page is loaded and becomes the current source of a parent Frame.
         /// </summary>
-        /// <param name="sender"></param>
         /// <param name="e"></param>
-        private void NavigateToSkeletalDetectorPage(object sender, RoutedEventArgs e)
+        protected override void OnNavigatedTo(NavigationEventArgs e)
         {
-            this.Frame.Navigate(typeof(SkeletalDetectorPage));
+            if (SkillPages.Count == 0)
+            {
+                LoadAllSkills();
+            }
+            base.OnNavigatedTo(e);
         }
 
         /// <summary>
-        /// Navigation to Object Detector Page
+        /// Load all the skills available
         /// </summary>
-        /// <param name="sender"></param>
-        /// <param name="e"></param>
-        private void NavigateToObjectDetectorPage(object sender, RoutedEventArgs e)
+        private async void LoadAllSkills()
+        {
+            // Task: bind category to UI tab
```

Comment:
>Task: [](start = 15, length = 5)

nitL not sure if you usually put task #, this one I think is 22514227

Reply:
Task 22514227: Load category tabs in MainPage using template & data binding

---
In reply to: [300711852](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/45#discussion_r300711852) [](ancestors = 300711852)

Reply:
@lobourre@microsoft.com  is it ok to add tfs ids to open source code?

---
In reply to: [300788037](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/45#discussion_r300788037) [](ancestors = 300788037,300711852)

Reply:
yeah it's fine, nothing to conceal here and it helps us track TODOs (these comments are public as well  btw ;) )

---
In reply to: [302176629](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/45#discussion_r302176629) [](ancestors = 302176629,300788037,300711852)

Reply:
got it thanks :)

---
In reply to: [302253598](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/45#discussion_r302253598) [](ancestors = 302253598,302176629,300788037,300711852)